libs/components/datetime/testing/src/date-range-picker/date-range-picker-harness.ts

```diff
@@ -0,0 +1,194 @@
+    return (await this.#getCalculatorSelectInput()).clickHelpInline();
+  }
+
+  /**
+   * Gets the help popover content.
+   */
+  public async getHelpPopoverContent(): Promise<string | undefined> {
+    return (await (
+      await this.#getCalculatorSelectInput()
+    ).getHelpPopoverContent()) as string | undefined;
+  }
+
+  /**
+   * Gets the help inline popover title.
+   */
+  public async getHelpPopoverTitle(): Promise<string | undefined> {
+    return await (await this.#getCalculatorSelectInput()).getHelpPopoverTitle();
+  }
+
+  /**
+   * Gets the hint text.
+   */
+  public async getHintText(): Promise<string> {
+    return await (await this.#getCalculatorSelectInput()).getHintText();
+  }
+
+  /**
+   * Gets the label text.
+   */
+  public async getLabelText(): Promise<string> {
+    return (await this.#getCalculatorSelectInput()).getLabelText();
+  }
+
+  /**
+   * Whether date range picker end date before start date error is thrown.
+   */
+  public async hasEndDateBeforeStartDateError(): Promise<boolean> {
+    return (await this.#getCalculatorSelectInput()).hasCustomFormError(
+      'endDateBeforeStartDate',
+    );
+  }
+
+  /**
+   * Whether end date input has required error.
+   */
+  public async hasEndDateRequiredError(): Promise<boolean> {
+    return (await this.#getEndDateInput()).hasRequiredError();
+  }
+
+  /**
+   * Whether a custom error has fired.
+   * @params errorName `errorName` property of the custom `sky-form-error`.
+   */
+  public async hasError(errorName: string): Promise<boolean> {
+    return (await this.#getCalculatorSelectInput()).hasCustomFormError(
+      errorName,
+    );
+  }
+
+  /**
+   * Whether start date input has required error.
+   */
+  public async hasStartDateRequiredError(): Promise<boolean> {
+    return (await this.#getStartDateInput()).hasRequiredError();
+  }
+
+  /**
+   * Whether the date range picker component is disabled.
+   */
+  public async isDisabled(): Promise<boolean> {
+    return (await this.#getCalculatorSelectInput()).getDisabled();
+  }
+
+  /**
+   * Whether end date datepicker is visible.
+   */
+  public async isEndDateVisible(): Promise<boolean> {
+    return !(await (
+      await this.locatorFor('.sky-date-range-picker-end-date')()
+    ).getProperty('hidden'));
+  }
+
+  /**
+   * Whether the date range picker has stacked enabled.
+   */
+  public async isStacked(): Promise<boolean> {
+    return (await this.host()).hasClass('sky-form-field-stacked');
+  }
+
+  /**
+   * Whether start date datepicker is visible.
+   */
+  public async isStartDateVisible(): Promise<boolean> {
+    return !(await (
+      await this.locatorFor('.sky-date-range-picker-start-date')()
+    ).getProperty('hidden'));
+  }
+
+  /**
+   * Selects the specified calculator.
+   */
+  public async selectCalculator(
+    calculatorId: SkyDateRangeCalculatorId,
+  ): Promise<void> {
+    const select = await this.#getCalculatorSelect();
+    return select.selectOptions(calculatorId);
```

Comment:
Check all methods and make sure they're returning the result of the awaited promise, instead of the promise itself. This will give us a proper error stack when debugging issues.
```
// Instead of...
return select.selectOptions(calculatorId);
// Do...
return await select.selectOptions(calculatorId);
```
Take a look at [this StackBlitz](https://stackblitz.com/edit/vitejs-vite-z4xrwj?file=main.js&terminal=dev) which demonstrates the issue.